tasks/retrieval_deal.go

```diff
@@ -174,3 +198,43 @@ func (rp *_RetrievalTask__Prototype) Of(minerParam string, payloadCid string, ca
 	}
 	return &rt
 }
+
+// ParseStageTimeouts parses "StageName=timeout" strings into a map of stage
+// name to timeout duration.
+func ParseStageTimeouts(timeoutSpecs []string) (map[string]time.Duration, error) {
+	var stageTimeouts map[string]time.Duration
+
+	// Parse all stage timeout durations
+	timeouts := map[string]time.Duration{}
+	for _, spec := range timeoutSpecs {
+		parts := strings.SplitN(spec, "=", 2)
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("invalid stage timeout specification: %s", spec)
+		}
+		stage := parts[0]
+		timeout := parts[1]
+		d, err := time.ParseDuration(strings.TrimSpace(timeout))
+		if err != nil || d < time.Second {
+			return nil, fmt.Errorf("invalid value for stage %q timeout: %s", stage, timeout)
+		}
+		timeouts[strings.ToLower(strings.TrimSpace(stage))] = d
+	}
+
+	// Get retrieval timeouts
+	for stageName, _ := range RetrievalStages {
+		stageName = strings.ToLower(stageName)
+		d, ok := timeouts[stageName]
+		if ok {
+			if stageTimeouts == nil {
+				stageTimeouts = make(map[string]time.Duration)
+			}
+			stageTimeouts[stageName] = d
+		}
+	}
```

Comment:
hmm, so if I make a typo in specifying a stage name, it won't be seen by the iteration over RetrievalStages and it will be ignored?

I think you should error if any input stage timeouts end up being unused because we don't know any stage by that name.

Reply:
not all deal flows will go through all stages / paths i think, which may complicate that

Reply:
Will return error if a stage timeout is unusable.  It is still possible that some deals may not use some stage timeouts, but this will at least report any user input that cannot be used.

Done